pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/instance/InstanceAssignmentDriver.java

```diff
@@ -60,19 +60,31 @@ public InstancePartitions assignInstances(InstancePartitionsType instancePartiti
         InstanceAssignmentConfigUtils.getInstanceAssignmentConfig(_tableConfig, instancePartitionsType);
     return getInstancePartitions(
         instancePartitionsType.getInstancePartitionsName(TableNameBuilder.extractRawTableName(tableNameWithType)),
-        assignmentConfig, instanceConfigs, existingInstancePartitions);
+        assignmentConfig, instanceConfigs, existingInstancePartitions, null);
+  }
+
+  public InstancePartitions assignInstances(InstancePartitionsType instancePartitionsType,
```

Comment:
nit: can the existing `assignInstances(InstancePartitionsType instancePartitionsType,
      List<InstanceConfig> instanceConfigs, @Nullable InstancePartitions existingInstancePartitions)` call this function since the remaining setup is the same? just to avoid too much code duplication